test/common/storagelib.py

```diff
@@ -554,8 +554,13 @@ def encrypt_root(self, passphrase):
 grubby --update-kernel=ALL --args="root=UUID=$uuid rootflags=defaults rd.luks.uuid=$luks_uuid rd.lvm.lv=root/root"
 ! test -f /etc/kernel/cmdline || cp /etc/kernel/cmdline /new-root/etc/kernel/cmdline
 """, timeout=300)
-        m.spawn("dd if=/dev/zero of=/dev/vda bs=1M count=100; reboot", "reboot", check=False)
-        m.wait_reboot(300)
+
+        # Destroy our current boot device.  This prevents the system from
+        # shutting down properly, so do it forcefully from outside.
+        m.spawn("sync; dd if=/dev/zero of=/dev/vda bs=1M count=100; sync; sync; sync;")
```

Comment:
You want m.execute() for that to make it sync. Will also fix [this failure](https://cockpit-logs.us-east-1.linodeobjects.com/pull-20449-3ac228be-20240508-064614-rhel-8-10-distropkg-storage/log.html#88)